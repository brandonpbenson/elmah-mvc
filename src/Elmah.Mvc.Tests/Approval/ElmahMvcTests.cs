using System.Text.RegularExpressions;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Elmah.Mvc.Tests.Approval
{
    [UseReporter(typeof(DiffReporter))]
    public class ElmahMvcTests
    {
        private readonly string ElmahMvcAppUrl = "http://localhost:49800";

        [Test]
        public void should_open_elmah_page(string path)
        {
            // do
            var url = ElmahMvcAppUrl + path;

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_stylesheets()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/stylesheet";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/rss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_digest_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/digestrss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_details()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_not_found_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_json_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/json?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_xml_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/xml?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        private string GetContent(string url)
        {
            return HackServertTime(ApprovalTests.Asp.AspApprovals.GetUrlContents(url));
        }

        private string HackServertTime(string content)
        {
            return new Regex("Server time is \\d+:\\d+:\\d+").Replace(content, string.Empty);
        }
    }
}
