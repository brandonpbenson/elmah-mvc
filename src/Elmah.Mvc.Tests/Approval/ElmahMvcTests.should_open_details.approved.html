<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
	<head><base href="http://localhost:49800/admin/elmah">
		<meta http-equiv="X-UA-Compatible" content="IE=EmulateIE7" />
		<title>
			Error: System.Web.HttpException [d5b2db50-0d5b-4c1d-a4dc-30324081191a]
		</title>
		<link rel="stylesheet" type="text/css" href="/admin/elmah/stylesheet" />

	</head>
	<body>
		<h1 id="PageTitle">
			A public action method &#39;st&#39; was not found on controller &#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&#39;.
		</h1>
		<ul id="SpeedList">
			<li><a href="/admin/elmah" title="List of logged errors">Errors</a></li><li><a href="http://elmah.googlecode.com/" title="Documentation, discussions, issues and more">Help</a></li><li><a href="/admin/elmah/about" title="Information about this version and build">About</a></li>
		</ul><p id="ErrorTitle"><span id="ErrorType">System.Web.HttpException</span><span id="ErrorTypeMessageSeparator">: </span><span id="ErrorMessage">A public action method &#39;st&#39; was not found on controller &#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&#39;.</span></p>
		<pre id="ErrorDetail">
System.Web.HttpException (0x80004005): A public action method &#39;st&#39; was not found on controller &#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&#39;.
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.Controller</span>.<span class='st-method'>HandleUnknownAction</span><span class='st-params'>(<span class='st-param-type'>String</span> <span class='st-param-name'>actionName</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.Controller</span>.<span class='st-method'>ExecuteCore</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.ControllerBase</span>.<span class='st-method'>Execute</span><span class='st-params'>(<span class='st-param-type'>RequestContext</span> <span class='st-param-name'>requestContext</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.ControllerBase.System.Web.Mvc.IController</span>.<span class='st-method'>Execute</span><span class='st-params'>(<span class='st-param-type'>RequestContext</span> <span class='st-param-name'>requestContext</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.MvcHandler.&lt;&gt;c__DisplayClass6.&lt;&gt;c__DisplayClassb</span>.<span class='st-method'>&lt;BeginProcessRequest&gt;b__5</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.Async.AsyncResultWrapper.&lt;&gt;c__DisplayClass1</span>.<span class='st-method'>&lt;MakeVoidDelegate&gt;b__0</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.Async.AsyncResultWrapper.&lt;&gt;c__DisplayClass8`1</span>.<span class='st-method'>&lt;BeginSynchronous&gt;b__7</span><span class='st-params'>(<span class='st-param-type'>IAsyncResult</span> <span class='st-param-name'>_</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1</span>.<span class='st-method'>End</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.MvcHandler.&lt;&gt;c__DisplayClasse</span>.<span class='st-method'>&lt;EndProcessRequest&gt;b__d</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.SecurityUtil</span>.<span class='st-method'>&lt;GetCallInAppTrustThunk&gt;b__0</span><span class='st-params'>(<span class='st-param-type'>Action</span> <span class='st-param-name'>f</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.SecurityUtil</span>.<span class='st-method'>ProcessInApplicationTrust</span><span class='st-params'>(<span class='st-param-type'>Action</span> <span class='st-param-name'>action</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.MvcHandler</span>.<span class='st-method'>EndProcessRequest</span><span class='st-params'>(<span class='st-param-type'>IAsyncResult</span> <span class='st-param-name'>asyncResult</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.Mvc.MvcHandler.System.Web.IHttpAsyncHandler</span>.<span class='st-method'>EndProcessRequest</span><span class='st-params'>(<span class='st-param-type'>IAsyncResult</span> <span class='st-param-name'>result</span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep</span>.<span class='st-method'>Execute</span><span class='st-params'>(<span class='st-param'></span>)</span></span>
   at <span class='st-frame'><span class='st-type'>System.Web.HttpApplication</span>.<span class='st-method'>ExecuteStep</span><span class='st-params'>(<span class='st-param-type'>IExecutionStep</span> <span class='st-param-name'>step</span>, <span class='st-param-type'>Boolean&amp;</span> <span class='st-param-name'>completedSynchronously</span>)</span></span>
		</pre>
		<p id="ErrorLogTime">Logged on 8 С‡РµСЂРІРЅСЏ 2012 СЂ. at 18:19:02</p>
		<p>See also:</p>
		<ul>
			<li>Raw/Source data in <a href="xml?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a" rel="alternate" type="application/xml">XML</a> or in <a href="json?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a" rel="alternate" type="application/json">JSON</a></li>
		</ul><div id="ServerVariables">
			<p class="table-caption">Server Variables</p>
			<div class="scroll-view">
				<table cellspacing="0">
					<tr>
						<th class="name-col" nowrap="nowrap">Name</th><th class="value-col" nowrap="nowrap">Value</th>
					</tr><tr class="even-row">
						<td class="key-col">ALL_HTTP</td><td class="value-col">HTTP_HOST:localhost:49800
</td>
					</tr><tr class="odd-row">
						<td class="key-col">ALL_RAW</td><td class="value-col">Host: localhost:49800
</td>
					</tr><tr class="even-row">
						<td class="key-col">APPL_MD_PATH</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">APPL_PHYSICAL_PATH</td><td class="value-col">D:\Development\Projects\elmah.mvc.controller\src\Elmah.Mvc.Web\</td>
					</tr><tr class="even-row">
						<td class="key-col">AUTH_PASSWORD</td><td class="value-col">*****</td>
					</tr><tr class="odd-row">
						<td class="key-col">AUTH_TYPE</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">AUTH_USER</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">CERT_COOKIE</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">CERT_FLAGS</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">CERT_ISSUER</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">CERT_KEYSIZE</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">CERT_SECRETKEYSIZE</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">CERT_SERIALNUMBER</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">CERT_SERVER_ISSUER</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">CERT_SERVER_SUBJECT</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">CERT_SUBJECT</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">CONTENT_LENGTH</td><td class="value-col">0</td>
					</tr><tr class="odd-row">
						<td class="key-col">CONTENT_TYPE</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">GATEWAY_INTERFACE</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">HTTP_HOST</td><td class="value-col">localhost:49800</td>
					</tr><tr class="even-row">
						<td class="key-col">HTTPS</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">HTTPS_KEYSIZE</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">HTTPS_SECRETKEYSIZE</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">HTTPS_SERVER_ISSUER</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">HTTPS_SERVER_SUBJECT</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">INSTANCE_ID</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">INSTANCE_META_PATH</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">LOCAL_ADDR</td><td class="value-col">::1</td>
					</tr><tr class="even-row">
						<td class="key-col">LOGON_USER</td><td class="value-col">seekey-note\alexander.beletsky</td>
					</tr><tr class="odd-row">
						<td class="key-col">PATH_INFO</td><td class="value-col">/admin/elmah/st</td>
					</tr><tr class="even-row">
						<td class="key-col">PATH_TRANSLATED</td><td class="value-col">D:\Development\Projects\elmah.mvc.controller\src\Elmah.Mvc.Web\admin\elmah\st</td>
					</tr><tr class="odd-row">
						<td class="key-col">QUERY_STRING</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">REMOTE_ADDR</td><td class="value-col">::1</td>
					</tr><tr class="odd-row">
						<td class="key-col">REMOTE_HOST</td><td class="value-col">::1</td>
					</tr><tr class="even-row">
						<td class="key-col">REMOTE_PORT</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">REMOTE_USER</td><td class="value-col"></td>
					</tr><tr class="even-row">
						<td class="key-col">REQUEST_METHOD</td><td class="value-col">GET</td>
					</tr><tr class="odd-row">
						<td class="key-col">SCRIPT_NAME</td><td class="value-col">/admin/elmah/st</td>
					</tr><tr class="even-row">
						<td class="key-col">SERVER_NAME</td><td class="value-col">localhost</td>
					</tr><tr class="odd-row">
						<td class="key-col">SERVER_PORT</td><td class="value-col">49800</td>
					</tr><tr class="even-row">
						<td class="key-col">SERVER_PORT_SECURE</td><td class="value-col">0</td>
					</tr><tr class="odd-row">
						<td class="key-col">SERVER_PROTOCOL</td><td class="value-col">HTTP/1.1</td>
					</tr><tr class="even-row">
						<td class="key-col">SERVER_SOFTWARE</td><td class="value-col"></td>
					</tr><tr class="odd-row">
						<td class="key-col">URL</td><td class="value-col">/admin/elmah/st</td>
					</tr>
				</table>
			</div>

		</div>
		<p id="Footer"><span>Powered by <a href="http://elmah.googlecode.com/">ELMAH</a>, version 1.2.14706.955. Copyright (c) 2004, Atif Aziz. All rights reserved. Licensed under <a href="http://www.apache.org/licenses/LICENSE-2.0">Apache License, Version 2.0</a>. </span>Server date is Friday, 08 June 2012. . All dates and times displayed are in the FLE Daylight Time zone. This log is provided by the XML File-Based Error Log.</p>
	</body>

</html>
